src/connection/mod.rs

```diff
@@ -564,11 +631,71 @@ impl Connection {
         Ok(vec)
     }
 
-    fn pull(&mut self) -> Result<Option<Record>, MgError> {
+    fn pull(&mut self, n: i64) -> Result<(), MgError> {
+        let pull_status = match n {
+            0 => unsafe { bindings::mg_session_pull(self.mg_session, std::ptr::null_mut()) },
+            _ => unsafe {
+                let mg_map = bindings::mg_map_make_empty(1);
+                if mg_map.is_null() {
+                    self.status = ConnectionStatus::Bad;
+                    return Err(MgError::new(String::from("Unable to make pull map.")));
+                }
+                let mg_int = bindings::mg_value_make_integer(n);
```

Comment:
This is partially my mistake, but this is also leaked when the insertion succeeds, because `mgp_map` doesn't take ownership, if I am not mistaken. Sorry for the correct review comment.

Reply:
All good, I should have been more careful here + communicate what is going on..., but according to the `mgclient` code (I haven't checked the implementation): 
```
If the insertion is successful, a new \ref mg_string is constructed for
the storage of the key and the map takes ownership of \p value.
```
in the success case, the value shouldn't be destroyed here?

Reply:
True, I was confused for a sec: `mg_map_insert` vs `mgp_map_insert` :D